### ISSUE TITLE
I2S audio crashes (Dell XPS 13, 9343)  

### ISSUE THREAD

**Issue by cribari** [advancingu/XPS13Linux#43]
I have a Dell XPS 13 notebook (9343, bios A07). My model is that with Intel i7 processor, 3200 x 1800 screen resolution, touchscreen and Broadcom wireless. 

cribari@darwin4 ~ $ dmesg | grep "XPS 13"
[ 0.000000] DMI: Dell Inc. XPS 13 9343/0310JH, BIOS A07 11/11/2015

I am currently running Arch Linux / Gnome 3.20 + GDM / kernel 4.6.2. I have been experiencing I2S audio crashes since kernel 4.4 (I ran Fedora back then). Audio works for a while and then crashes. At that point dmesg informs: 

[24294.494682] haswell-pcm-audio haswell-pcm-audio: error: audio DSP boot timeout IPCD 0x0 IPCX 0x0
[24294.811357] haswell-pcm-audio haswell-pcm-audio: ipc: --message timeout-- ipcx 0x83000000 isr 0x00000000 ipcd 0x00000000 imrx 0x7fff0000
[24294.811360] haswell-pcm-audio haswell-pcm-audio: error: stream commit failed
[24294.811375] haswell-pcm-audio haswell-pcm-audio: ASoC: haswell-pcm-audio hw params failed: -110
[24295.118072] haswell-pcm-audio haswell-pcm-audio: ipc: --message timeout-- ipcx 0x83000000 isr 0x00000000 ipcd 0x00000000 imrx 0x7fff0000
[24295.118076] haswell-pcm-audio haswell-pcm-audio: error: stream commit failed
[24295.118082] haswell-pcm-audio haswell-pcm-audio: ASoC: haswell-pcm-audio hw params failed: -110

I believe the problem is caused by GDM (perhaps through pulseaudio). 

A dmesg dump is available at http://pastebin.com/D3iyt0GA

I filed this bug report https://bugs.archlinux.org/task/49556 but received no feedback so far. 

Suggestions are welcome.

**Comment by antonmry** [comment]
This is a very frustrating bug, lot of information here: https://bugzilla.redhat.com/show_bug.cgi?id=1313434

It's for fedora, but I think the problem it's the same with Archlinux. I have exactly the same computer and it's working after apply some of the workarounds explained in the thread, except the headphones mic. Yet it's very disappointing :-(

**Comment by AdmiralAsshat** [comment]
I have sound working now on my 9343 with 4.3+ kernels. I believe I had to manually unmute the SP0 channel in alsamixer to bring it back, as well as unmute two other channels to get headphones working.

My still unresolved issue, however, is that when using headphones, if there is no sound playing, I can hear feint pops coming through the headphones from time to time.

I'd still like that to be fixed, but at this point, I spent so much time and energy just getting sound to work that I'm willing to live with the quiet popping. I've usually got music streaming anyway.

**Comment by advancingu** [comment]
@AdmiralAsshat Try turning off energy saving for the audio components with Powertop. The popping usually occurs each time the audio chip is powered up or down.

**Comment by AdmiralAsshat** [comment]
@advancingu Thanks for the recommendation, but according to powertop, energy saving is already turned off. It's the entry "Enable Audio codec power management", right?

**Comment by advancingu** [comment]
Yes but there is also `Runtime PM for PCI Device Intel Corporation Broadwell-U Audio Controller` which is disabled on my system as well (`bad` status).

**Comment by AdmiralAsshat** [comment]
It's set to "Bad" for me as well.

**Comment by patrl** [comment]
@AdmiralAsshat  I grappled with the headphone popping for weeks and was never able to find a fix. If possible, I'd recommend recompiling the kernel with the following flag set: `CONFIG_ACPI_REV_OVERRIDE_POSSIBLE=y`. The sound card will then be loaded in HDA mode and work perfectly. I doubt the issues with I2S on this model will be fixed upstream anytime soon.
